theano/gpuarray/ROIPoolGPUFwd.c

```diff
@@ -0,0 +1,120 @@
+                    hend = min(max(hend + roi_start_h, 0), height);
+                    wstart = min(max(wstart + roi_start_w, 0), width);
+                    wend = min(max(wend + roi_start_w, 0), width);
+                    bool is_empty = (hend <= hstart) || (wend <= wstart);
+                    ga_int pool_index = ph * pooled_width + pw;
+                    if (is_empty) {
+                        channel_out[pool_index] = 0;
+                        channel_argmax[pool_index] = -1;
+                    }
+                    // Define an empty pooling region to be zero
+                    // If nothing is pooled, argmax = -1 causes nothing to be backprop'd
+                    for (ga_int h = hstart; h < hend; ++h) {
+                        for (ga_int w = wstart; w < wend; ++w) {
+                            ga_int bottom_index = h * width + w;
+                            if (channel_data[bottom_index] > channel_out[pool_index]) {
+                                channel_out[pool_index] = channel_data[bottom_index];
+                                channel_argmax[pool_index] = bottom_index;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+#section support_code_struct
+
+bool vector_same_shape(PyGpuArrayObject* arr1, PyGpuArrayObject* arr2){
+  return (PyGpuArray_DIMS(arr1)[0] == PyGpuArray_DIMS(arr2)[0]);
+  }
+
+int APPLY_SPECIFIC(ROIPoolGPUFwd)(PyGpuArrayObject *data,
+                      PyGpuArrayObject *rois,
+                      PyGpuArrayObject **argmaxes,
+		              PyGpuArrayObject **out,
+                      PyGpuContextObject *ctx) {
+    size_t address = 1;
+    int num_rois = PyGpuArray_DIMS(rois)[0];
+    int batch_size =  PyGpuArray_DIMS(data)[0];
+    int channels = PyGpuArray_DIMS(data)[1];
+    int height = PyGpuArray_DIMS(data)[2];
+    int width = PyGpuArray_DIMS(data)[3];
+
+    // Prepare outputs.
+    int err;
+
+    if (!GpuArray_IS_C_CONTIGUOUS(&data->ga) || !GpuArray_IS_C_CONTIGUOUS(&rois->ga)){
+        PyErr_Format(PyExc_ValueError, "GpuRoIPoolOp: requires data to be C-contiguous");
+        return 1;
+    }
+
+    if (*out != NULL || *argmaxes != NULL || (!vector_same_shape(data, *out)) || (!vector_same_shape(data, *argmaxes))){
+    Py_XDECREF(*out);
+    Py_XDECREF(*argmaxes);
+    size_t dim[4];
+    dim[0] = batch_size;
+    dim[1] = num_rois;
+    dim[2] = channels;
+    dim[3] = POOLED_HEIGHT * POOLED_WIDTH;
+    *out = (PyGpuArrayObject*) pygpu_zeros(4, dim, data->ga.typecode, GA_C_ORDER, ctx, Py_None);
+    *argmaxes = (PyGpuArrayObject*) pygpu_zeros(4, dim, data->ga.typecode, GA_C_ORDER, ctx, Py_None);
+    if (*out == NULL || *argmaxes == NULL) {
+      PyErr_Format(PyExc_ValueError, "Could not allocate output storage");
+      return 1;
+    }
+  }
+
+  err = ROIPoolGPUFwd_kernel_scall(1, &address, 0,
+          num_rois, data->ga.data, SPATIAL_SCALE, channels, height, width,
```

Comment:
@nouiz I have a problem here. When i try to print SPATIAL_SCALE here, it is 1. or whatever the value I pass from the tests. But in the kernel, that is always 0.00. Could you please help me in finding the fault ?